tests/cli/test_parsers.py

```diff
@@ -0,0 +1,181 @@
+                help="The user or organization the repo is under."
+            ),
+            call(
+                'repo',
+                help="The repository to remove a hook from."
+            )
+        ]
+
+        expected_set_defaults_calls = [
+            call(func=remove_hook)
+        ]
+
+        add_unregister_command(commands)
+
+        remove_command_parser.add_argument.assert_has_calls(expected_add_argument_calls)
+        remove_command_parser.set_defaults.assert_has_calls(expected_set_defaults_calls)
+
+    def test_add_org_register_command_adds_register_org_hook_invocation(self):
+        commands = Mock()
+        org_register_command_parser = Mock()
+
+        desc = (
+            "Register webhook for a given organization\n"
+            "The installed webhook will be used to trigger lint\n"
+            "reviews as pull requests are opened/updated.\n"
+        )
+
+        commands.add_parser = conditionally_return(
+            org_register_command_parser,
+            'org-register',
+            help=desc
+        )
+
+        expected_add_argument_calls = [
+            call(
+                '-u',
+                '--user',
+                dest='login_user',
+                help=(
+                    "The OAuth token of the user that has admin rights to the org "
+                    "you are adding hooks to. Useful when the user "
+                    "in settings is not the administrator of "
+                    "your organization."
+                )
+            ),
+            call(
+                'org_name',
+                help="The login name of the organization."
+            )
+        ]
+
+        expected_set_defaults_calls = [
+            call(func=register_org_hook)
+        ]
+
+        add_org_register_command(commands)
+
+        org_register_command_parser.add_argument.assert_has_calls(expected_add_argument_calls)
+        org_register_command_parser.set_defaults.assert_has_calls(expected_set_defaults_calls)
+
+    def test_add_org_unregister_command_adds_remove_org_hook_invocation(self):
+        commands = Mock()
+        org_unregister_command_parser = Mock()
+
+        desc = "Unregister webhooks for a given organization.\n"
+
+        commands.add_parser = conditionally_return(
+            org_unregister_command_parser,
+            'org-unregister',
+            help=desc
+        )
+
+        expected_add_argument_calls = [
+            call(
+                '-u', '--user',
+                dest='login_user',
+                help=(
+                    "The OAuth token of the user that has admin rights to the org "
+                    "you are removing hooks from. Useful when the "
+                    "user in settings is not the administrator of "
+                    "your organization."
+                )
+            ),
+            call(
+                'org_name',
+                help="The login name of the organization."
+            )
+        ]
```

Comment:
These tests seem to duplicate most of the logic in the commands. Is there a way we could use `responses` to instead do an integration test?

Reply:
Thanks for your feedback Mark - can you give me an example of what you might see as an integration test flow?  In my view, it seems like we'd need something like a mock GitHub api server.